pkg/trace/api/api.go

```diff
@@ -43,6 +42,18 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
+var bufferPool = sync.Pool{
+	New: func() interface{} {
+		return new(bytes.Buffer)
+	},
+}
+
+func newBuffer() *bytes.Buffer {
```

Comment:
Can we go one step further and have it be `popBuffer` and `pushBuffer`. That'll make it less error prone and might address your concern about forgetting to put it back...